test/e2e/test_gcp_broker.go

```diff
@@ -51,83 +48,60 @@ Note: the number denotes the sequence of the event that flows in this test case.
 */
 
 func GCPBrokerTestImpl(t *testing.T, authConfig lib.AuthConfig) {
```

Comment:
FYI, this may have a small conflict with #1022 which adds the same metrics-checking logic to this test impl as is in the broker_pubsub test.

Reply:
Yep. There will be a conflict since this PR did a refactor of the duplicate code between GCP Broker and Broker with PubSub Channel. But I think that should be quite fast to resolve the conflicts.

Reply:
I solved the merge conflicts. But you may need to add more metrics-checking for the new E2E tests added in this PR once this PR got merged.